### DISCUSSION TITLE
Chats

### DISCUSSION THREAD

**Discussion by belgishboy** [GQDeltex/ft_transcendence#52] [Ideas]
So, from the way I read things, You can have either Channels or Direct messages.
Channels
- can be either **public**(visible) or **private**(not visible/require invite).
- Can **protected** (have a password).
- Channel creator is **Channel Owner**.
          - can set a password, remove it, or modify it.
- **Channel Owner** is also an **Administrator**
- **Administrators** can add more **Administrators**
          - Can **ban** or **mute** users for a limited time.

**Comment by belgishboy** [comment]
To be determined:
-Can Admins ban/mute/remove other admins or can only the Owner do that?
-Can Owner leave the channel (who becomes channel owner if they do)? (I would say they cant because it make our lives so much easier)
-Do you need to enter the password if you are invited to a chat ?
-Who can invite to a chat (Owner/Admin/User)?

**Reply by GQDeltex** [reply]
I think it is stated nowhere that you should be able to invite people into a chat. So I'd just not implement the whole invitation thing, which would mean we don't have to worry about if we ask for password or not. There is only JOIN and that asks for a password
